### PR TITLE
chore: Migrate to AndroidX and Koin 1.X

### LINKED ISSUE #766: Migrate to AndroidX
Migrate to AndroidX dependencies

### PR DESCRIPTION
**Pull request by iamareebjamal**
Fixes #766

### CODE REVIEW

**Comment by simarsingh24** [comment]
Why build is failing?

File: app/build.gradle [hunk 1]
```diff
@@ -6,14 +6,14 @@ apply plugin: 'kotlin-android-extensions'
 def STRIPE_API_TOKEN = System.getenv('STRIPE_API_TOKEN') ?: "YOUR_API_KEY"
 
 android {
-    compileSdkVersion 27
+    compileSdkVersion 28
```
Comment (nikit19) [inline app/build.gradle]: I guess travis.yml needs to be updated as well

**Comment by iamareebjamal** [comment]
```
before_install:
- yes | sdkmanager "platforms;android-28"
```

This is already there. But I'll check

**Comment by iamareebjamal** [comment]
OK, so first it installs SDK 28, then overwrites it with 27. Great!

**Comment by iamareebjamal** [comment]
Also SearchFragment was not updated

**Pull request merged**